### PR TITLE
Have .asd files follow recommended ASDF practice

### PR DESCRIPTION
**Pull request by fare**
Most importantly, stop calling clear-system from perform. BAD BAD BAD.

Use "strings" as canonical system names.

Do not define long description by default, but for those who want to read it
from a file (not recommended), do it a simpler way.

### CODE REVIEW

**Comment by fare** [comment]
PING

**Comment by ramonkeypr** [comment]
@fare, does your branch need to update update the Travis CI script, or are there changes that need to occur in your code to make the tests pass?

https://travis-ci.org/fukamachi/caveman/jobs/215426458

**Comment by fare** [comment]
I recently sent fixes to `prove` that were merged. Can you restart this test with the latest prove?

**Comment by fare** [comment]
As for me, when I manually run `(asdf:test-system :caveman2-test)` or `(asdf:test-system :caveman-test)` from my local patched setup with quicklisp loaded, I get no error — which I *believe* should be what the tests should do, though I don't understand the ins and outs enough to swear I didn't break the test system somehow (e.g. it says `Summary: All 0 files passed.`).

File: caveman-test.asd [hunk 1]
```diff
@@ -6,24 +6,17 @@
...
-               :cl-test-more
-               :uiop
-               :dexador
-               :usocket)
+(defsystem "caveman-test"
+  :depends-on ("caveman"
+               "cl-test-more"
+               "uiop"
+               "dexador"
+               "usocket")
```
Comment (ramonkeypr) [inline caveman-test.asd]: What's the advantage of using strings over a symbol or an atom?

**Comment by fare** [comment]
See explanations in https://gitlab.common-lisp.net/asdf/asdf/blob/master/doc/best_practices.md

**Comment by fare** [comment]
NB: destroyed https://github.com/fare/caveman and instead created https://github.com/fare-patches/caveman

File: caveman.asd [hunk 2]
```diff
@@ -51,19 +46,7 @@
...
-        (let ((seq (make-array (file-length stream)
-                               :element-type 'character
-                               :fill-pointer t)))
-          (setf (fill-pointer seq) (read-sequence seq stream))
-          seq)))
-  :in-order-to ((test-op (test-op caveman-test))))
-
-(defmethod asdf:perform :after ((op asdf:load-op) (system (eql (asdf:find-system :caveman))))
-  (warn "\"caveman\" is deprecated now. Did you mean \"caveman2\"?"))
+  :long-description #.(read-file-string (subpathname *load-pathname* "README.v1.markdown"))
```
[the remark this answers is not in the document]
Comment (fare) [reply caveman.asd]: A `.asd` file is loaded by `load-asd` which binds `*package*` to `asdf-user`. Whichever way you are evaluating the file seems dubious.

File: caveman-test.asd (hunk 1 above)
Comment (fare) [reply caveman-test.asd]: It's canonical.

**Comment by fare** [comment]
This PR is superseded by #110. Closing.

**Pull request closed without merging**